internal/mage/sdk/python.go

```diff
@@ -0,0 +1,87 @@
+package sdk
+
+import (
+	"context"
+	"os"
+
+	"dagger.io/dagger"
+	"github.com/dagger/dagger/internal/mage/util"
+	"github.com/magefile/mage/mg"
+)
+
+var _ SDK = Go{}
+
+type Python mg.Namespace
+
+// Lint lints the Python SDK
+func (t Python) Lint(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "lint:style"},
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+// Test tests the Python SDK
+func (t Python) Test(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "test"},
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+// Generate re-generates the SDK API
+func (t Python) Generate(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "generate"},
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+func pythonBase(c *dagger.Client) *dagger.Container {
+	repo := util.Repository(c)
+
+	pyproject := c.Directory().
+		WithFile("pyproject.toml", repo.File("sdk/python/pyproject.toml")).
+		WithFile("README.md", repo.File("sdk/python/README.md"))
+	return c.Container().From("python:3.10-alpine").
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"apk", "add", "-U", "--no-cache", "gcc", "musl-dev", "libffi-dev"},
+		}).
+		WithEnvVariable("PIP_NO_CACHE_DIR", "off").
+		WithEnvVariable("PIP_DISABLE_PIP_VERSION_CHECK", "on").
+		WithEnvVariable("PIP_DEFAULT_TIMEOUT", "100").
+		WithWorkdir("/app/sdk/python").
+		WithMountedDirectory("/app/sdk/python", pyproject).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"pip", "install", "hatch"},
+		}).
+		WithEnvVariable("HATCH_ENV", "default").
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "env", "create"},
+		}).
```

Comment:
You should add another exec for `hatch env create lint`.

Reply:
What about `test`?

Reply:
`test` is on the default environment.

Reply:
The reason for the split in envs is that the default installs dagger, but the lint env doesn't. Because it uses tools outside of the need to import dagger things. But I keep considering putting it all in the same env because of CI.

Reply:
Done!